code/src/main/java/com/onlinecontacttracing/servlets/delete-data/NegativeUserPlacesDeleter.java

```diff
@@ -0,0 +1,60 @@
+package com.onlinecontacttracing.servlets;
+
+import static com.googlecode.objectify.ObjectifyService.ofy;
+import com.onlinecontacttracing.storage.NegativeUserPlace;
+import com.onlinecontacttracing.storage.Constants;
+import com.google.common.collect.Iterables;
+import com.googlecode.objectify.Key;
+import java.util.logging.Logger;
+import java.time.Instant;
+
+/**
+* This class deletes old negative user places
+*/
+class NegativeUserPlacesDeleter extends OldDataDeleter {
+
+  public Class getType() {
+    return NegativeUserPlace.class;
+  }
+
+  public String getQueryString() {
+    return "place.timeInterval.intervalStartSeconds <";
+  }
+
+  /**
+  * To calculate the time for querying old data, this method
+  * subtracts maximum time a class is allowed to persist in memory
+  * from the current time by the 
+  */
+  public long getMaxAge() {
+    return Instant.now().getEpochSecond() - Constants.NEGATIVE_USER_DATA_MAX_TIME;
+  }
+
+  /**
+  * This method deletes old data.
+  * If the query was successful, the number of entries deleted will be logged.
+  * Otherwise the exception is caught and logged.
+  */
+  public void delete(Logger log) {
```

Comment:
This method can be in the OldDataDeleter class, that's the purpose of the inheritance :)

Reply:
Makes sense, I think I wasn't zooming out and that's why I was hesitant to make the changes but this is clearly better. Sorry for the reluctance.